### PR TITLE
Updates to Bankbot to work with skip bank

### PR DESCRIPTION
**Pull request by intangibles**
updated bankbot changes

old PR #1752 w/ comments

### CODE REVIEW

File: common-money.lic [hunk 1]
```diff
@@ -28,8 +30,16 @@ module DRCM
     amount
   end
 
+  def hometown_currency(hometown_name)
+    return currency = get_data('town')[hometown_name]['currency']
```
Comment (rpherbig) [inline common-money.lic]: You don't need to assign to `currency` here (or even use an explicit `return`)

File: common-money.lic [hunk 2]
```diff
@@ -28,8 +30,16 @@ module DRCM
...
+  end  
+
   def wealth(hometown)
-    currency = get_data('town')[hometown]['currency']
+    if $CURRENCIES.join("|").match /#{hometown}/i
```
Comment (rpherbig) [inline common-money.lic]: Does this mean we're calling `wealth` with an arg of a currency sometimes and a hometown sometimes?

File: sell-loot.lic [hunk 3]
```diff
@@ -19,24 +19,45 @@ class SellLoot
...
 
-    return if settings.sell_loot_skip_bank
-    exchange_coins
+    return if settings.sell_loot_skip_bank && !bankbot_enabled
+
+    @keep_coppers = parse(settings.sell_loot_money_on_hand.split(' '))
+    @keep_coppers = @bankbot_deposit_threshold if bankbot_enabled
+
+    if bankbot_enabled
+      if @bankbot_name.nil? || @bankbot_room_id.nil?
```
Comment (rpherbig) [inline sell-loot.lic]: Is this more appropriate to have in `validate.lic`?

File: sell-loot.lic [hunk 4]
```diff
@@ -19,24 +19,45 @@ class SellLoot
...
-    check_bankbot(settings.hometown, keep_copper)
+    if settings.sell_loot_skip_bank && bankbot_enabled
+      $CURRENCIES.each{|currency|
+        give_money_to_bankbot(currency, currency == @local_currency ? @keep_coppers : 0)
+      }
+      return
+    end
+
+    exchange_coins
+    if bankbot_enabled == false
```
Comment (rpherbig) [inline sell-loot.lic]: I recommend reversing the order of the conditionals, so the `if true` clause is first.

File: sell-loot.lic [hunk 5]
```diff
@@ -60,30 +81,26 @@ class SellLoot
...
+      .reject { |currency| currency =~ /#{exchange_to}/i }
       .each { |currency| fput "exchange all #{currency} for #{exchange_to}" }
   end
 
   def deposit_coins(keep_copper)
     walk_to @hometown['deposit']['id']
     release_invisibility
     bput('wealth', 'Wealth:')
-    case bput('deposit all', 'you drop all your', 'You hand the clerk some coins', "You don't have any", 'There is no teller here', 'reached the maximum balance I can permit')
+    case bput('deposit all', 'you drop all your', 'You hand the clerk some coins', "You don't have any", 'There is no teller here')
```
Comment (rpherbig) [inline sell-loot.lic]: Why was this line changed?

File: sell-loot.lic [hunk 6]
```diff
@@ -19,24 +19,45 @@ class SellLoot
...
     check_spare_pouch(settings.spare_gem_pouch_container, settings.gem_pouch_adjective) if settings.spare_gem_pouch_container
 
     sell_bundle if settings.sell_loot_bundle
 
-    return if settings.sell_loot_skip_bank
-    exchange_coins
+    return if settings.sell_loot_skip_bank && !bankbot_enabled
+
+    @keep_coppers = parse(settings.sell_loot_money_on_hand.split(' '))
+    @keep_coppers = @bankbot_deposit_threshold if bankbot_enabled
```
Comment (rpherbig) [inline sell-loot.lic]: If `bankbot_enabled` is true, should we instead keep the max of `settings.sell_loot_money_on_hand` and `@bankbot_deposit_threshold`?

Comment (rpherbig) [reply sell-loot.lic]: If not, maybe we should have an entry in `validate.lic` that says you have `sell_loot_money_on_hand` set, but also `bankbot_enabled` is true. Something to indicate that `sell_loot_money_on_hand` is ignored.

File: common-money.lic (hunk 2 above)
Comment (intangibles) [reply common-money.lic]: Yeah, there are a variety of places that use hometown name.  Either by calling wealth directly or indirectly via ensure_copper_on_hand, or other methods I haven't seen yet.

For example:

crossing-repair:
ensure_copper_on_hand(@repair_withdrawal_amount, settings.hometown)

deposit.lic:
copper_on_hand = wealth(hometown)

File: sell-loot.lic (hunk 3 above)
Comment (intangibles) [reply sell-loot.lic]: Possibly, there were checks to return if those settings were missing in the method give_money_to_bankbot.  I figured it would be more efficient to move them to the top so we don't go through any methods if they are missing.

Would you rather have me have exit the method w/o posting an error message?

File: common-money.lic (hunk 2 above)
Comment (rpherbig) [reply common-money.lic]: Could we have two methods (and the one that looks up your hometown currency could call the other one)?

File: sell-loot.lic (hunk 3 above)
Comment (rpherbig) [reply sell-loot.lic]: Or we can assume that if they have `bankbot_enabled`, they filled in both name and id?

File: sell-loot.lic (hunk 5 above)
Comment (intangibles) [reply sell-loot.lic]: Whoops, I had a rebase mishap so used my old code on updated master and didn't notice this change.

I'll update.

File: common-money.lic (hunk 1 above)
Comment (intangibles) [reply common-money.lic]: Updated

File: sell-loot.lic (hunk 3 above)
Comment (intangibles) [reply sell-loot.lic]: Yeah, I'm okay with that.

File: sell-loot.lic (hunk 4 above)
Comment (intangibles) [reply sell-loot.lic]: updated

File: sell-loot.lic (hunk 6 above)
Comment (intangibles) [reply sell-loot.lic]: Ideally I'd like to have one setting.  How would you feel above removing bankbot_deposit_threshhold?

File: common-money.lic (hunk 2 above)
Comment (intangibles) [reply common-money.lic]: Err... deposit.lic is not a dependency script. it was a copy of something I was using before.  If that matters.  It may be easier to clean up the scripts that call the method instead.

File: sell-loot.lic (hunk 6 above)
Comment (intangibles) [reply sell-loot.lic]: So... after further review on this.  Both settings are needed because bank withdraw uses "1 silver" and bankbot uses "tip 100".  They don't mix very well.  (Or at all)

I recommend they stay independent and people can just figure it out.  One goes for bank withdrawals, one is for bank deposit.

Bankbot deposit threshhold only works for the local currency based on current hometown settings. (assuming you are deposting all currencies.)  I considered added the option to keep multiple currencies but decided it would be not be used.

**Comment by intangibles** [comment]
re: deposit threshholds
 
So... after further review on this. Both settings are needed because bank withdraw uses "1 silver" and bankbot uses "tip 100". They don't mix very well. (Or at all)

I recommend they stay independent and people can just figure it out. One goes for bank withdrawals, one is for bank deposit.

Bankbot deposit threshhold only works for the local currency based on current hometown settings. (assuming you are deposting all currencies.) I considered added the option to keep multiple currencies but decided it would be not be used.

**Comment by intangibles** [comment]
re: deposit thresholds part 2.

Something else that has been bothering me about bankbot_deposit_threshhold.  I actually never use this because if you keep have a threshhold of "1 plat", you could end up keeping 5,000 coppers.  Instead, i set all deposits to zero and have a small withdrawal script for any pieces i want to keep.

Not sure how others use it.

File: common-money.lic [hunk 7]
```diff
@@ -29,18 +29,17 @@ module DRCM
   end
 
   def hometown_currency(hometown_name)
-    return currency = get_data('town')[hometown_name]['currency']
+    currency = get_data('town')[hometown_name]['currency']
```
Comment (rpherbig) [inline common-money.lic]: Does `currency` need to be assigned here?

**Comment by rpherbig** [comment]
@rcuhljr any comments?

File: common-money.lic (hunk 7 above)
Comment (intangibles) [reply common-money.lic]: Ahh. No it doesn't.  Thanks!
